urlchecker/client/__init__.py

```diff
@@ -141,6 +141,60 @@ def get_parser():
         default=5,
     )
 
+    # main check_ci entrypoint
+
+    check_ci = subparsers.add_parser(
+        "check_ci", help="check urls in the specified files"
+    )
+
+    check_ci.add_argument("files", metavar="file", nargs="+")
```

Comment:
Why should check_ci have all these arguments if none of them can be set when the hook is run? The only argument that you seem to be parsing is files, and the rest are set to defaults (or edited below). That would make more sense I think for this use case.

Reply:
All of the options `check_ci` takes are used and not defaulted (beyond the argparse defaults). force-pass,  no-print, exclude-urls, save, retry-count, and timeout can be used.

Repo owners can override the 'entry' parameter and use the options in their `.pre-commit-config.yaml` file. You're correct that the default entry point in `.pre-commit-hooks.yaml` doesn't use them.

I don't see this mode only being used by CI systems.

Reply:
okay, if the arguments are shared then you can do something like:

```python
for command in check_ci, check:
    # add common command here.
```
so we don't have the redundancy.

Reply:
Ah, that is nice. Done.